protocol-units/bridge/contracts/test/AtomicBridgeCounterpartyMOVE.t.sol

```diff
@@ -180,89 +191,152 @@ contract AtomicBridgeCounterpartyMOVETest is Test {
-    vm.startPrank(deployer); // The deployer is the owner
-    atomicBridgeCounterpartyMOVE.abortBridgeTransfer(bridgeTransferId);
-
-    (
-        bytes32 abortedInitiator,
-        address abortedRecipient,
-        uint256 abortedAmount,
-        bytes32 abortedHashLock,
-        uint256 abortedTimeLock,
-        AtomicBridgeCounterpartyMOVE.MessageState abortedState
-    ) = atomicBridgeCounterpartyMOVE.bridgeTransfers(bridgeTransferId);
-
-    assertEq(abortedInitiator, initiator);
-    assertEq(abortedRecipient, recipient);
-    assertEq(abortedAmount, amount);
-    assertEq(abortedHashLock, hashLock);
-    assertLe(
-        abortedTimeLock,
-        block.timestamp,
-        "Timelock is not less than or equal to current timestamp"
-    );
-    assertEq(
-        uint8(abortedState),
-        uint8(AtomicBridgeCounterpartyMOVE.MessageState.REFUNDED)
-    );
-
-    vm.stopPrank();
-}
+    function testRateLimitExceeded() public {
+        uint256 moveAmount = 100 * 10**8; // 100 MOVEToken
+        moveToken.transfer(originator, moveAmount); 
+
+        vm.startPrank(originator);
+        moveToken.approve(address(atomicBridgeInitiatorMOVE), moveAmount);
+
+        // Initiate the first bridge transfer through the initiator
+        atomicBridgeInitiatorMOVE.initiateBridgeTransfer(
+            moveAmount / 2,
+            initiator,
+            hashLock
+        );
 
+        vm.stopPrank();
+
+        // Start locking the transfer on the counterparty side
+        vm.startPrank(deployer);
 
+        // Lock the first transfer (within rate limit)
+        atomicBridgeCounterpartyMOVE.lockBridgeTransfer(
+            initiator,
+            bridgeTransferId,
+            hashLock,
+            recipient,
+            moveAmount / 2
+        );
+
+        // Verify the first transfer was locked successfully
+        (
+            bytes32 lockedInitiator,
+            address lockedRecipient,
+            uint256 lockedAmount,
+            bytes32 lockedHashLock,
+            uint256 lockedTimeLock,
+            AtomicBridgeCounterpartyMOVE.MessageState lockedState
+        ) = atomicBridgeCounterpartyMOVE.bridgeTransfers(bridgeTransferId);
+        assertEq(lockedInitiator, initiator);
+        assertEq(lockedRecipient, recipient);
+        assertEq(lockedAmount, moveAmount / 2);
+        assertEq(lockedHashLock, hashLock);
+        assertGt(lockedTimeLock, block.timestamp);
+        assertEq(
+            uint8(lockedState),
+            uint8(AtomicBridgeCounterpartyMOVE.MessageState.PENDING)
+        );
+
+        // Attempt a second transfer to exceed the rate limit
+        bytes32 secondBridgeTransferId = keccak256(
+            abi.encodePacked(block.timestamp, initiator, recipient, moveAmount, hashLock, timeLock)
+        );
+
+        // Expect the second transfer to exceed the rate limit and revert
+        vm.expectRevert("RATE_LIMIT_EXCEEDED");
+        atomicBridgeCounterpartyMOVE.lockBridgeTransfer(
+            initiator,
```

Comment:
1. I'd like to see tests demonstrating both the L1->L2 direction and the L2->L1 direction.
2. For user experience, L1->L2 should be able to revert on `initiateBridgeTransfer`, no?
3. In general, why don't we use the `bridgeTransferId` returned by initiate?